File: app/src/main/java/huce/fit/appreadstories/account/manager/AccountManagerImpl.java
package huce.fit.appreadstories.account.manager;

import android.content.Context;
import android.util.Log;

import huce.fit.appreadstories.account.BaseAccountImpl;

public class AccountManagerImpl extends BaseAccountImpl implements AccountManagerPresenter {

    private static final String TAG = "AccountManagerImpl";
    AccountManagerImpl(AccountManagerView accountManagerView, Context context) {
        super(context);
        Log.e(TAG, "AccountManagerImpl");
        accountManagerView.setName(getSharedPreferences().getName());
    }

    @Override
    public void logout() {
        setSharedPreferences().myRemove();
    }
}

File: app/src/main/java/huce/fit/appreadstories/account/manager/AccountManagerImpl.kt
package huce.fit.appreadstories.account.manager

import android.content.Context
import huce.fit.appreadstories.account.BaseAccountImpl

class AccountManagerImpl(accountManagerView: AccountManagerView, context: Context): BaseAccountImpl(context) , AccountManagerPresenter {

    companion object{
        const val TAG = "AccountManagerImpl"
    }
    init{
        accountManagerView.setName(getAccount().getName()!!)
    }

    override fun logout() {
        setAccount().myRemove()
    }
}
